lib/_tls_wrap.js

```diff
@@ -383,13 +395,6 @@ TLSSocket.prototype._init = function(socket, wrap) {
   if (options.handshakeTimeout > 0)
     this.setTimeout(options.handshakeTimeout, this._handleTimeout);
 
-  // Socket already has some buffered data - emulate receiving it
-  if (socket && socket._readableState && socket._readableState.length) {
-    var buf;
-    while ((buf = socket.read()) !== null)
-      ssl.receive(buf);
-  }
-
```

Comment:
Why do it in both places, wouldn't doing it once in `_init` complete the same job?